src/app.rs

```diff
@@ -467,20 +467,16 @@ pub fn build_runtime(threads: Option<usize>, thread_name: &str) -> Result<Runtim
     rt_builder.max_blocking_threads(20_000);
     rt_builder.enable_all().thread_name(thread_name);
 
-    if let Some(threads) = threads {
-        if threads < 1 {
-            #[allow(clippy::print_stderr)]
-            {
-                eprintln!("The `threads` argument must be greater or equal to 1.");
-            }
-            return Err(exitcode::CONFIG);
-        } else {
-            WORKER_THREADS
-                .set(NonZeroUsize::new(threads).expect("already checked"))
-                .expect("double thread initialization");
-            rt_builder.worker_threads(threads);
-        }
+    let threads = threads.unwrap_or_else(crate::num_threads);
+    if threads < 1 {
+        error!("The `threads` argument must be greater or equal to 1.");
+        return Err(exitcode::CONFIG);
     }
+    WORKER_THREADS
+        .set(NonZeroUsize::new(threads).expect("already checked"))
+        .expect("double thread initialization");
+    rt_builder.worker_threads(threads);
+    info!("tokio runtime created with {threads} worker threads");
```

Comment:
I feel like we probably don't need to mention `tokio` here, and I'm a little bit hesitant to log it at `info` in the first place. I can definitely see how it could be helpful for debugging, but there's also the potential for confusion given that this isn't the actual total number of threads users should expect to see.

Reply:
I agree this is more suitable for a debug log. Updated.